src/bootstrap/src/lib.rs

```diff
@@ -315,46 +315,19 @@ impl Build {
         let in_tree_llvm_info = config.in_tree_llvm_info.clone();
         let in_tree_gcc_info = config.in_tree_gcc_info.clone();
 
-        let initial_target_libdir_str = if config.dry_run() {
-            "/dummy/lib/path/to/lib/".to_string()
-        } else {
-            output(
-                Command::new(&config.initial_rustc)
-                    .arg("--target")
-                    .arg(config.build.rustc_target_arg())
-                    .arg("--print")
-                    .arg("target-libdir"),
-            )
-        };
+        let initial_target_libdir_str =
+            config.initial_sysroot.join("lib/rustlib").join(config.build).join("lib");
+
         let initial_target_dir = Path::new(&initial_target_libdir_str).parent().unwrap();
         let initial_lld = initial_target_dir.join("bin").join("rust-lld");
 
-        let initial_sysroot = if config.dry_run() {
-            "/dummy".to_string()
-        } else {
-            output(Command::new(&config.initial_rustc).arg("--print").arg("sysroot"))
-        }
-        .trim()
-        .to_string();
-
-        // FIXME(Zalathar): Determining this path occasionally fails locally for
-        // unknown reasons, so we print some extra context to help track down why.
-        let find_initial_libdir = || {
-            let initial_libdir =
-                initial_target_dir.parent()?.parent()?.strip_prefix(&initial_sysroot).ok()?;
-            Some(initial_libdir.to_path_buf())
-        };
-        let Some(initial_libdir) = find_initial_libdir() else {
-            panic!(
-                "couldn't determine `initial_libdir`:
-- config.initial_rustc:      {rustc:?}
-- initial_target_libdir_str: {initial_target_libdir_str:?}
-- initial_target_dir:        {initial_target_dir:?}
-- initial_sysroot:           {initial_sysroot:?}
-",
-                rustc = config.initial_rustc,
-            );
-        };
+        let initial_libdir = initial_target_dir
+            .ancestors()
+            .nth(2)
+            .unwrap()
+            .strip_prefix(&config.initial_sysroot)
+            .unwrap()
+            .to_path_buf();
```

Comment:
I have never experienced this failing before and it should be even more stable now.

cc @Zalathar

Reply:
I still see this fail occasionally (on macOS only), but I have already collected plenty of examples of the failure so I don’t really need this code any more.

If you replace the unwrap with something like `.expect("couldn’t determine initial libdir")`, I’m fine with that.

Reply:
For reference, the failure looks like this: https://github.com/rust-lang/rust/pull/130138#issuecomment-2402056712

Two different invocations of the same compiler binary produce different sysroot paths. It probably has something to do with hardlink bookkeeping on macOS specifically.

Reply:
Oh I see above that we aren’t invoking rustc multiple times anymore, so yeah from a bootstrap perspective this should avoid the problem I was seeing.

Reply:
> If you replace the unwrap with something like .expect("couldn’t determine initial libdir"), I’m fine with that.

Done that in https://github.com/rust-lang/rust/compare/16f30f2f2d92ee450e4ec715ee09e17dc2c1f5b8..27323aac9fdb905e0665fc9c43147a895ef51da6.